Mage.Sets/src/mage/cards/c/CaptainAmericaFirstAvenger.java

```diff
@@ -0,0 +1,215 @@
+import mage.abilities.effects.common.DamageMultiEffect;
+import mage.constants.*;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.common.FilterEquipmentPermanent;
+import mage.filter.predicate.ObjectSourcePlayer;
+import mage.filter.predicate.ObjectSourcePlayerPredicate;
+import mage.filter.predicate.permanent.AttachedToPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.Target;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetAnyTargetAmount;
+
+/**
+ *
+ * @author Grath
+ */
+public final class CaptainAmericaFirstAvenger extends CardImpl {
+
+    private static final FilterPermanent filter2 = new FilterEquipmentPermanent("equipment you control");
+
+    static {
+        filter2.add(TargetController.YOU.getControllerPredicate());
+    }
+
+    public CaptainAmericaFirstAvenger(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{R}{W}{U}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.subtype.add(SubType.HERO);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Throw ... — {3}, Unattach an Equipment from Captain America: He deals damage equal to that Equipment’s mana value divided as you choose among one, two, or three targets.
+        Ability ability = new SimpleActivatedAbility(
+                new DamageMultiEffect(CaptainAmericaFirstAvengerValue.instance).setText(
+                        "he deals damage equal to that Equipment's mana value divided as you choose among one, two, or three targets."),
+                new GenericManaCost(3));
+        ability.addCost(new CaptainAmericaFirstAvengerUnattachCost());
+        ability.addTarget(new TargetAnyTargetAmount(CaptainAmericaFirstAvengerValue.instance, 3));
+        this.addAbility(ability.withFlavorWord("Throw ..."));
+
+        // ... Catch — At the beginning of combat on your turn, attach up to one target Equipment you control to Captain America.
+        ability = new BeginningOfCombatTriggeredAbility(
+                new CaptainAmericaFirstAvengerCatchEffect(), TargetController.YOU, false
+        );
+        ability.addTarget(new TargetPermanent(0, 1, filter2));
+        this.addAbility(ability.withFlavorWord("... Catch"));
+    }
+
+    private CaptainAmericaFirstAvenger(final CaptainAmericaFirstAvenger card) {
+        super(card);
+    }
+
+    @Override
+    public CaptainAmericaFirstAvenger copy() {
+        return new CaptainAmericaFirstAvenger(this);
+    }
+}
+
+enum CaptainAmericaPredicate implements ObjectSourcePlayerPredicate<MageObject> {
+    instance;
+
+    // Functional negation of AnotherPredicate.
```

Comment:
would it work to use `Predicates.not(AnotherPredicate.instance)`?

Reply:
IntelliJ throws a red line about not being the right type when I try that.